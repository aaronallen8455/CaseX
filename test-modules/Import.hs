module Import
  ( Foo(..)
  ) where

data Foo = Foo | Bar | Baz
