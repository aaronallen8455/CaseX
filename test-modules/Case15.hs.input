{-# LANGUAGE OrPatterns #-}
import           Import

main :: IO ()
main = pure ()

test :: Maybe Foo -> Bool
test x = case x of
  (Nothing; Just (Foo; Bar; SPLIT)) -> True

test2 :: Maybe (Maybe Foo) -> Bool
test2 x = case x of
  (Just (SPLIT; Nothing); SPLIT) -> True
