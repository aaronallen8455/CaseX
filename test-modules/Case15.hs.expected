{-# LANGUAGE OrPatterns #-}
import           Import

main :: IO ()
main = pure ()

test :: Maybe Foo -> Bool
test x = case x of
  (Nothing; Just (Foo; Bar)) -> True
  Just Baz -> True

test2 :: Maybe (Maybe Foo) -> Bool
test2 x = case x of
  (Just ( Nothing)) -> True
  Nothing -> True
  Just (Just _) -> True
